TestVectorsAwsCryptographicMaterialProviders/runtimes/python/fuzz_generator.py

```diff
@@ -0,0 +1,338 @@
+        "reproducedEncryptionContext": reproduced_context,
+        "requiredEncryptionContextKeys": required_keys,
+        "encryptionContext": encryption_context
+    }
+    
+    # Add error descriptions for negative tests
+    add_error_descriptions(test_vector, test_type, keyring_type)
+    
+    return test_vector
+
+
+def extract_new_keys(test_vectors: Dict[str, Any]) -> Dict[str, Any]:
+    """Extract new keys from raw keyring test vectors.
+
+    - Scans all generated test vectors
+    - Finds raw keyring tests (type == "raw")
+    - Extracts the fuzzed key names they reference
+    - Creates corresponding key material entries for keys.json
+    - Returns a dict of new keys to add to keys.json
+    """
+    new_keys = {}
+    
+    for test_vector in test_vectors.values():
+        if test_vector.get("encryptKeyDescription", {}).get("type") == "raw":
+            key_name = test_vector["encryptKeyDescription"]["key"]
+            
+            # Determine base key type from key name
+            base_key_id = "aes-256"  # default
+            for key_type in RAW_KEY_TYPES:
+                if key_type in key_name:
+                    base_key_id = key_type
+                    break
+            
+            key_info = KEY_MATERIALS.get(base_key_id, KEY_MATERIALS["aes-256"])
+            new_keys[key_name] = {
+                "encrypt": True, "decrypt": True, "algorithm": "aes", "type": "symmetric",
+                "bits": key_info["bits"], "encoding": "base64",
+                "material": key_info["material"], "key-id": base_key_id
+            }
+    
+    return new_keys
+
+
+def generate_fuzz_test_vectors(num_vectors: int = 5) -> Tuple[Dict[str, Any], Dict[str, Any]]:
+    """Generate multiple fuzzed test vectors and collect new key generated."""
+    test_vectors = {}
+    
+    for i in range(num_vectors):
+        test_vector = fuzz_test_vector().example()
+        test_id = str(uuid.uuid4())
+        test_vectors[test_id] = test_vector
+    
+    new_keys = extract_new_keys(test_vectors)
+    return test_vectors, new_keys
+
+
+def main():
+    """Main function to generate fuzzed test vectors."""
+    # Generate test vectors and new keys
+    test_vectors, new_keys = generate_fuzz_test_vectors(num_vectors=5)
+    
+    # Load and update keys.json
+    try:
+        with open("keys.json", "r") as f:
+            keys_data = json.load(f)
+    except FileNotFoundError:
+        print("Error: keys.json not found!")
+        return
+    
+    original_key_count = len(keys_data["keys"])
+    keys_data["keys"].update(new_keys)
+    
+    with open("keys.json", "w") as f:
+        json.dump(keys_data, f, indent=2, ensure_ascii=False)
+    
+    # Create and save manifest.json
+    manifest_data = {
+        "manifest": {"version": 4, "type": "awses-mpl-encrypt"},
+        "keys": "file://keys.json",
+        "tests": test_vectors
+    }
+    
+    with open("manifest.json", "w") as f:
```

Comment:
This might not dump the JSON in right place for all language. Add a TODO for this.